providers/ldap/ldap_client.go

```diff
@@ -173,7 +174,11 @@ func (l *LClient) GenerateToken(jsonInput map[string]string) (model.Token, int,
 		query,
 		searchConfig.UserSearchAttributes, nil)
 
-	return l.userRecord(search, lConn)
+	token, status, err := l.userRecord(search, lConn)
+	if l.AccessMode == "required" && status == 403 {
+		return token, status, errors.New("Account does not have access")
+	}
+	return token, status, err
```

Comment:
I think it is better to do a separate search when l.AccessMode == "required" and conclude that the status to be thrown is 403, than doing the checks in the userRecord() utility method